### PR TITLE
Fix `use_onnx` parameter usage in `from_pretrained` func and update `test_download_no_onnx_by_default` test

### PR DESCRIPTION
**Pull request by JacquiML**
# What does this PR do?

- Fixes https://github.com/huggingface/diffusers/pull/4338#issuecomment-1664402373. 
- Also update `test_download_no_onnx_by_default` func, split that function to two test cases `by default no onnx weights are downloaded for non-ONNX pipelines` and `by default onnx weights are downloaded for ONNX pipelines`

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [ ] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.
@patrickvonplaten

### CODE REVIEW

File: src/diffusers/utils/dummy_torch_and_transformers_and_onnx_objects.py [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 class OnnxStableDiffusionImg2ImgPipeline(metaclass=DummyObject):
     _backends = ["torch", "transformers", "onnx"]
+    _is_onnx = True
```
Comment (DN6) [inline src/diffusers/utils/dummy_torch_and_transformers_and_onnx_objects.py:7]: I don't think we need this hard code this flag. When running these tests with `onnxruntime` installed will result in this flag being set to true.

File: src/diffusers/utils/import_utils.py [hunk 2]
```diff
@@ -570,6 +570,8 @@ class DummyObject(type):
     `requires_backend` each time a user tries to access any method of that class.
     """
 
+    _is_onnx = False
```
Comment (DN6) [inline src/diffusers/utils/import_utils.py:573]: This flag doesn't need to be hard coded. It won't be present when trying to create `OnnxPipelines` without the `onnxruntime` installed. So an error message will be thrown.

File: tests/pipelines/test_pipelines.py [hunk 3]
```diff
@@ -327,28 +327,28 @@ def test_download_no_openvino_by_default(self):
     def test_download_no_onnx_by_default(self):
         with tempfile.TemporaryDirectory() as tmpdirname:
             tmpdirname = DiffusionPipeline.download(
-                "hf-internal-testing/tiny-random-OnnxStableDiffusionPipeline",
+                "hf-internal-testing/tiny-stable-diffusion-pipe-indexes",
```
Comment (DN6) [inline tests/pipelines/test_pipelines.py:330]: I don't think there are any `onnx` files present in this. So this test isn't verifying that the onnx files are being ignored by default. You can try using 
this repo `hf-internal-testing/tiny-stable-diffusion-xl-pipe`

File: tests/pipelines/test_pipelines.py [hunk 4]
```diff
@@ -327,28 +327,28 @@ def test_download_no_openvino_by_default(self):
...
                 cache_dir=tmpdirname,
+                use_safetensors=False,
             )
 
             all_root_files = [t[-1] for t in os.walk(os.path.join(tmpdirname))]
             files = [item for sublist in all_root_files for item in sublist]
 
-            # make sure that by default no onnx weights are downloaded
+            # make sure that by default no onnx weights are downloaded for not ONNX pipelines
```
Comment (DN6) [inline tests/pipelines/test_pipelines.py:338]:
```suggestion
            # make sure that by default no onnx weights are downloaded for non-ONNX pipelines
```

File: tests/pipelines/test_pipelines.py [hunk 5]
```diff
@@ -327,28 +327,28 @@ def test_download_no_openvino_by_default(self):
...
                 "hf-internal-testing/tiny-random-OnnxStableDiffusionPipeline",
                 cache_dir=tmpdirname,
-                use_onnx=True,
             )
 
             all_root_files = [t[-1] for t in os.walk(os.path.join(tmpdirname))]
             files = [item for sublist in all_root_files for item in sublist]
 
-            # if `use_onnx` is specified make sure weights are downloaded
+            # make sure that by default onnx weights are downloaded for ONNX pipelines
```
Comment (DN6) [inline tests/pipelines/test_pipelines.py:351]: I think it might be better to move this into it's own test with a `require_onnxruntime` decorator.

**Comment by DN6** [comment]
Hi @JacquiML. Thank you for taking the time to put this together. I've left some comments on your PR. Additionally, I noticed a couple of files were committed here `pipeline_alt_diffusion` and `pipeline_alt_diffusion_img2img`. These aren't relevant to the ONNX issue. 

Could you please open a separate PR with only the relevant changes?

**Comment by JacquiML** [comment]
> Hi @JacquiML. Thank you for taking the time to put this together. I've left some comments on your PR. Additionally, I noticed a couple of files were committed here `pipeline_alt_diffusion` and `pipeline_alt_diffusion_img2img`. These aren't relevant to the ONNX issue.
> 
> Could you please open a separate PR with only the relevant changes?

Hey @DN6 , thank you for your PR review and comments! I have addressed some of them, and am working on the left one. Also revert the file changes not related to this onnx stuff/PR.

**Comment by DN6** [comment]
Hi @JacquiML. Let me know if you need any assistance. I think once you address this comment:
https://github.com/huggingface/diffusers/pull/4508/files#r1293761196 

We should be ready to merge

**Comment by JacquiML** [comment]
> Hi @JacquiML. Let me know if you need any assistance. I think once you address this comment: https://github.com/huggingface/diffusers/pull/4508/files#r1293761196
> 
> We should be ready to merge

Hey @DN6 , the comment has been addressed, could you please have another review? Thank you!

**Review by DN6 — APPROVED** [review]
LGTM 👍🏽

**Comment by JacquiML** [comment]
> LGTM 👍🏽

Thank you @DN6! But I don't have access to merge this PR, right?

**Comment by DN6** [comment]
Merged! Great job @JacquiML 🔥

**Comment by JacquiML** [comment]
> Merged! Great job @JacquiML 🔥

Awesome! Thank you for the support along the way @DN6 ! :+1:

**Pull request merged**